### PR TITLE
BATM-1844 verification_site module serving customer web for identity verification

### PR DESCRIPTION
**Pull request by kubacech**

### CODE REVIEW

**Review by generalbytes — CHANGES_REQUESTED** [review]
Remove all mentions of Onfido. 
This website should be verification service agnostic.
how about verification_site?

**Comment by kubacech** [comment]
> Remove all mentions of Onfido.
> This website should be verification service agnostic.
> how about verification_site?

module renamed to verification_site

File: verification_site/src/main/resources/static/script.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            console.log("onfido wizard finished. Notifying server.")
+            let xhttp = new XMLHttpRequest()
+            xhttp.open("GET", "/verification/submit/" + applicantId, true)
+            xhttp.send()
+        },
+        steps: [
+            {
+                type: 'welcome',
+                options: {
+                    title: 'Verify Your Identity',
```
Comment (generalbytes) [inline verification_site/src/main/resources/static/script.js:20]: Add support for czech language too.

File: verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private final RestTemplate restTemplate;
+    private static final Logger log = LoggerFactory.getLogger(VerificationController.class);
+
+    public VerificationController(ApplicantService applicantService, RestTemplate restTemplate) {
+        this.applicantService = applicantService;
+        this.restTemplate = restTemplate;
+    }
+
+    @PostMapping(value = "/register", consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity register(@RequestBody RegisterNewApplicantReq req) {
```
Comment (generalbytes) [inline verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java:33]: Add javadoc. Mainly who call this method and why.

File: verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        applicantService.addApplicant(req);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping("/token/{applicant}")
```
Comment (generalbytes) [inline verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java:38]: Add javadoc. Mainly who call this method and why.

File: verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    public ResponseEntity getSdkToken(@PathVariable("applicant") String applicantId) {
+        Applicant applicant = applicantService.get(applicantId);
+        if (applicant != null) {
+            return ResponseEntity.ok(applicant.getSdkToken());
+        }
+        return ResponseEntity.status(NOT_FOUND).build();
+    }
+
+    @GetMapping("/submit/{applicant}")
```
Comment (generalbytes) [inline verification_site/src/main/java/com/generalbytes/verification/controller/VerificationController.java:47]: Add javadoc. Mainly who call this method and why.

File: verification_site/src/main/java/com/generalbytes/verification/service/ApplicantService.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import java.time.Instant;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+@Service
+public class ApplicantService {
+
+    private static final Long TIMEOUT_DURATION_SECONDS = 90 * 60L; // every SDK token is valid only for 90 minutes
+    private final ConcurrentHashMap<String, Applicant> CACHE = new ConcurrentHashMap<>();
```
Comment (generalbytes) [inline verification_site/src/main/java/com/generalbytes/verification/service/ApplicantService.java:16]: Use CacheBuilder.

Example is here: https://github.com/GENERALBYTESCOM/batm_public/blob/8f76ef5697f1a5887d98c88e324607b4e9a1270e/server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/XChangeExchange.java#L69

**Review by generalbytes — CHANGES_REQUESTED** [review]
Add also description how to compile, configure and run verification_site in existing README.md
Thank you.

**Comment by generalbytes** [comment]
Add also description how to compile, configure and run verification_site in existing README.md
Thank you.

**Comment by kubacech** [comment]
I updated PR with requested changes.

**Pull request merged**